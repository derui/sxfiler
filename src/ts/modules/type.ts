import { Action } from "redux";

export type AppAction<T extends string, Extra extends {} = {}> = Action<T> & {
  payload: { [K in keyof Extra]: Extra[K] };
};

// generate type for actions with action implementations and action types
export type ActionsType<T extends { [key: string]: string }, ActionCreators extends object> = {
  [key in keyof ActionCreators]: ActionCreators[key] extends (...args: any[]) => AppAction<T[keyof T], any>
    ? ReturnType<ActionCreators[key]>
    : never;
}[keyof ActionCreators];
