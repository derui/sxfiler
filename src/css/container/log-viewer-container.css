.log-viewer-container__root {
  position: relative;
  height: 100%;
  width: 100%;
  max-width: 100%;
  border-top: 1px solid var(--color-logViewer-separator);
}
