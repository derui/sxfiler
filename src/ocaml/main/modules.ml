module N = Jsoo_node
module FFI = Sxfiler_common.Ffi

(** Call require and return module. This function should use for nodejs's module. *)
let require module_ : 'a Js.t =
  let require = Js.Unsafe.pure_js_expr "require" in
  let module_ = Js.string module_ in
  Js.Unsafe.(fun_call require [|inject module_|])

let fs () : N.Module_types.fs Js.t = require "fs"
let original_fs () : N.Module_types.fs Js.t = require "original-fs"

let electron () = require "electron"

let crash_reporter () : FFI.Crash_reporter.t Js.t =
  Js.Unsafe.get electron "crashReporter"

let browser_window: FFI.BrowserWindow.option Js.t -> FFI.BrowserWindow.t Js.t = fun option ->
  let cls = Js.Unsafe.get electron "BrowserWindow" in
  new%js cls option
