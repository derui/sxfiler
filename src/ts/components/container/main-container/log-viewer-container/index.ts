export * from "./log-viewer-container";
