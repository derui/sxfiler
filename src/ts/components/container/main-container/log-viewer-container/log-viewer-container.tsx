import * as React from "react";
import { styled } from "@/components/theme";

import * as Element from "@/components/ui/element";
import { State } from "@/states/log-entry";
import { Component as LogViewer } from "@/components/project/log-viewer";

export type Props = {
  state: State;
};

const Root = styled(Element.Component)`
  position: relative;

  height: 100%;
  width: 100%;
  max-width: 100%;
  border-top: 1px solid ${props => props.theme.colors.base2};
`;

export const Component: React.FC<Props> = ({ state }) => {
  return (
    <Root>
      <LogViewer entries={Array.from(state.entries.values())} hidden={false} />
    </Root>
  );
};
