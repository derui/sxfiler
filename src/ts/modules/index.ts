//#IMPORT INDICATOR
import * as Configuration from "./configuration";
import * as Keymap from "./keymap";
import * as Filer from "./filer";
import * as Completer from "./completer";
import * as Decision from "./decision";
// prettier-ignore
import { combineReducers } from "redux";

// export combined types
// prettier-ignore
export type Actions =
  | Configuration.Actions
  | Keymap.Actions
  | Filer.Actions
  | Completer.Actions
  | Decision.Actions
  ;

// export combined reducer
// prettier-ignore
export const reducer = combineReducers({
  configuration: Configuration.reducer,
  keymap: Keymap.reducer,
  filer: Filer.reducer,
  completer: Completer.reducer,
  decision: Decision.reducer,
})

export const emptyState = {
  configuration: Configuration.emptyState,
  keymap: Keymap.emptyState,
  filer: Filer.emptyState,
  completer: Completer.emptyState,
  decision: Decision.emptyState,
};

// export state type
export type State = ReturnType<typeof reducer>;
