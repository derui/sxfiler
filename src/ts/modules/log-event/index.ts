import { Actions as Actions$ } from "./actions";
import { State as State$, LogEvents as LogEvents_ } from "./reducer";

export type Actions = Actions$;
export type State = State$;

export { ActionTypes } from "./types";
export { actions } from "./actions";
export { reducer, emptyState, LogEventCreators } from "./reducer";

export type LogEvents = LogEvents_;
