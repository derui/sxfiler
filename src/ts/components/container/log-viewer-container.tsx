import { h } from "preact";
import { Component as LogViewer } from "@/components/project/log-viewer";
import { State } from "@/modules";

export type Props = {
  state: State;
};

export const Component: preact.FunctionComponent<Props> = ({ state }) => {
  return (
    <div class="log-viewer-container__root" data-testid="logViewerContainer">
      <LogViewer entries={state.logEvent.events} hidden={false} />
    </div>
  );
};
