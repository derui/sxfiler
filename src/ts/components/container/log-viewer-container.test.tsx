import { h } from "preact";
import { render } from "preact-render-to-string";
import { Component } from "./log-viewer-container";
import { emptyState } from "@/modules";

describe("Container", () => {
  describe("Log Viewer Container", () => {
    it("should render correctly", () => {
      const tree = render(<Component state={emptyState} />);

      expect(tree).toMatchSnapshot();
    });
  });
});
