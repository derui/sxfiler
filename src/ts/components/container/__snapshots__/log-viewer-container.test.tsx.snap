// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Log Viewer Container should render correctly 1`] = `"<div class=\\"log-viewer-container__root\\" data-testid=\\"logViewerContainer\\"><div class=\\"log-viewer__root\\" data-testid=\\"logViewer-root\\" aria-hidden=\\"false\\"></div></div>"`;
